lib/core.js

```diff
@@ -295,18 +295,26 @@ declare class JSON {
 }
 
 /* Iterators */
-interface IteratorResult<T> {
+interface IteratorResult<Y,R> {
     done: boolean;
-    value?: T;
+    value?: Y|R;
```

Comment:
So for `Iterator<string>`, you'll have `value?: string | void`. It's a little weird that there's a `void` due to being optional and a `void` due to there being no return, but I think it's fine. 

I actually wonder if maybe we can make value not be optional, since the union'd `void` captures the behavior perfectly. Like if iterate over a `Generator<string, number, void>`, will the `IteratorResult` ever be omitted? Looking at [the mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*) it seems like it shouldn't be `undefined` if there is a return